### PR TITLE
Bugfix

### PR DESCRIPTION
**Pull request by wenjdu**
Essentially review AppDelegate and View Controller for optimisation, less lines of code is the best 🍡

### CODE REVIEW

File: alps-ios-app-color/ViewController.swift [hunk 1]
```diff
@@ -9,46 +9,63 @@
...
+                  "teal": UIColor(red: 20/255, green: 229/255, blue: 204/255, alpha: 1),
+                  "pink": UIColor(red: 255/255, green: 173/255, blue: 171/255, alpha: 1),
+                  "white": UIColor(red: 255/255, green: 255/255, blue: 255/255, alpha: 1)
+    ]
     
     override func viewDidLoad() {
         super.viewDidLoad()
+        
+        // Start Monitoring Matches
+        self.onMatch = { matches, _ in
```
Comment (maciejburda) [inline alps-ios-app-color/ViewController.swift]: you are missing`[weak self]`

It's not a big deal it this scenario, but it's best practice.

File: alps-ios-app-color/ViewController.swift [hunk 2]
```diff
@@ -9,46 +9,63 @@
...
+            print("🏔 You've got new matches!!! 🏔\n\(matches.map { $0.publication?.encodeToJSON() })")
+            // Color the background according to the match
+            let mostRecentDate = matches.max(by: {
+                $0.createdAt! < $1.createdAt!
+            })
+            guard let properties = mostRecentDate?.publication?.properties else {
+                print("No properties.")
+                return
+            }
+            let color = properties["color"] as? String
```
Comment (maciejburda) [inline alps-ios-app-color/ViewController.swift]: you should also guard let color. Line below you have to force unwrap it with `!` which is not good practice.

File: alps-ios-app-color/ViewController.swift [hunk 3]
```diff
@@ -9,46 +9,63 @@
...
+            self.view.backgroundColor = self.colors[color!]
+        }
+        MatchMore.matchDelegates += self
+        
         // Do any additional setup after loading the view, typically from a nib.
         publishButton.layer.cornerRadius = 4
```
Comment (maciejburda) [inline alps-ios-app-color/ViewController.swift:54]: all these big chunks of code should be moved to separate function to make reading more clear.

File: alps-ios-app-color/ViewController.swift (hunk 1 above)
Comment (wenjdu) [reply alps-ios-app-color/ViewController.swift]: Weak self on onMatch variable ? I will add it thanks.

File: alps-ios-app-color/ViewController.swift (hunk 2 above)
Comment (wenjdu) [reply alps-ios-app-color/ViewController.swift]: got it !

File: alps-ios-app-color/ViewController.swift (hunk 3 above)
Comment (wenjdu) [reply alps-ios-app-color/ViewController.swift:54]: ok !

**Pull request merged**